http-webhook.md

```diff
@@ -0,0 +1,376 @@
+time or as a "pre-flight" request immediately preceding a delivery. If the
+handshake outcome is that the delivery is denied, the sender MUST NOT deliver
+events to the target.
+
+It is important to understand is that the handshake does not aim to establish
+an authentication or authorization context. It only serves to protect the
+sender from being told to a push to a destination that is not expecting the
+traffic. While this specification mandates use of an authorization model, this
+mandate is not sufficient to protect any arbitrary website from unwanted
+traffic if that website doesn't implement access control and therefore ignores
+the `Authorization` header.
+
+Delivery targets SHOULD support the abuse protection feature. If a target does
+not support the feature, the sender MAY choose not to send to the target, at
+all, or send only at a very low request rate.
+
+### 4.1. Validation request
+
+The validation request uses the HTTP [OPTIONS][OPTIONS] method. The request
+is directed to the exact resource target URI that is being registered.
+
+With the validation request, the sender asks the target for permission to
+send notifications, and it can declare a desired request rate (requests per
+minute).
+
+The delivery target will respond with a permission statement and the permitted
+request rate.
+
+The following header fields are for inclusion in the validation request.
+
+#### 4.1.2. WebHook-Request-Origin
+
+The `WebHook-Request-Origin` header MUST be included in the validation request
+and requests permission to send notifications from this sender, and contains a
+DNS expression that identifies the sending system, for example
+"eventemitter.example.com". The value is meant to summarily identify all sender
+instances that act on the behalf of a certain system, and not an individual
+host.
+
+After the handshake and if permission has been granted, the sender MUST use the
+`Origin` request header for each delivery request, with the value matching that
+of this header.
+
+Example:
+
+``` text 
+WebHook-Request-Origin: eventemitter.example.com 
+```
+
+#### 4.1.3. WebHook-Request-Callback
+
+The `WebHook-Request-Callback` header is OPTIONAL and augments the
+`WebHook-Request-Origin` header. It allows the delivery target to grant send
+permission asynchronously, via a simple HTTPS callback.
+
+If the receiving application does not explicitly support the handshake
+described here, an administrator could nevertheless still find the callback URL
+in the log, and call it manually and therewith grant access.
+
+The delivery target grants permission by issuing an HTTPS GET or POST request against
+the given URL. The HTTP GET request can be performed manually using a browser
+client.
+
+The delivery target MAY include the `WebHook-Allowed-Rate` response in the callback.
+
+The URL is not formally constrained, but it SHOULD contain an identifier for the
+delivery target along with a secret key that makes the URL difficult to guess so that
```

Comment:
Maybe suggest it should be temporary secret key.

Reply:
What do you think "temporary" adds that would otherwise be misunderstood by an implementer?

Reply:
IMO these tokens, just as whole subscription request if not confirmed through handshake, should be expiring after some time. Without mentioning these are temporary, it may be assumed that one can confirm at anytime in the future, subscription requests and tokens to be persisted endlessly until confirmed back.